lib/tests/test_commit_builder.rs

```diff
@@ -257,11 +268,15 @@ fn test_rewrite_resets_author_timestamp(backend: TestRepoBackend) {
 
     // Create discardable commit
     let initial_timestamp = "2001-02-03T04:05:06+07:00";
-    let config = testutils::base_config()
-        .set_override("debug.commit-timestamp", initial_timestamp)
-        .unwrap()
-        .build()
-        .unwrap();
+    let mut config = testutils::base_user_config();
+    config.add_layer(ConfigLayer::with_data(
+        ConfigSource::User,
+        config::Config::builder()
+            .set_override("debug.commit-timestamp", initial_timestamp)
+            .unwrap()
+            .build()
+            .unwrap(),
+    ));
```

Comment:
> Not all tests are ported to ConfigLayer::parse() because it seemed a bit odd
to format!() a TOML document and parse it to build a table of configuration
variables.

Maybe we'll want a `ConfigLayer::set_value()` so this can be something like this:

```
let mut layer = ConfigLayer::empty(ConfigSource::User);
layer.set_value("debug.commit-timestamp", initial_timestamp);
config.add_layer(layer);
```

Just an idea. Not asking for any changes in this PR.

Reply:
Yes, that's one option. Once migrated to `toml_edit`, we can initialize the variables in a similar way to `HashMap`. Maybe there will be some helper function to set a value with dotted name path.